src/metadata.rs

```diff
@@ -20,42 +20,28 @@ impl fmt::Display for Metadata {
     }
 }
 
-#[derive(Debug)]
-pub struct MetadataError {
-    message: String,
-}
+/// MetadataError enumerates all errors returned by metadata module.
+#[derive(thiserror::Error, Debug)]
+pub enum MetadataError {
+    /// Representation has multiple results and thus ambiguous.
+    #[error("Ambiguous date")]
+    AmbiguousDate,
 
-impl MetadataError {
-    fn from_str(message: &str) -> Self {
-        MetadataError {
-            message: message.to_string(),
-        }
-    }
+    /// Representation has invalid date.
+    #[error("Invalid date")]
+    InvalidDate,
 
-    fn from_string(message: String) -> Self {
-        MetadataError { message }
-    }
-}
+    /// Catch all errors when trying to parse date from exif.
```

Comment:
I think this comment is not correct. This error about the exif date not being ascii. Other parsing errors would probably result in an invalid date. For example, `100/10/10` cannot be parsed as date but it is ascii.

Reply:
Good catch. Improved it now.